samples: audio: sof: compilation issue, include file not found.

**Describe the bug**
While running automatic tests on STM32 test bench,
samples/audio/sof build fails to build.
This appends on all STM32 boards but could also be seen on NON STM32 (frdm_k64f)

This is a new sample introduced by #30336

**To Reproduce**
Steps to reproduce the behavior:
1. scripts/twister --ninja --runtime-artifact-cleanup -b  -p nucleo_f429zi  -T samples/audio/sof/ --outdir ./log_db_audio
2. See error

**Expected behavior**
Build successful or sample not build for those boards

**Logs and console output**
```
In file included from /home/osboxes/zephyr/zephyr_project/modules/audio/sof/zephyr/../src/arch/xtensa/include/arch/drivers/timer.h:13,
                 from /home/osboxes/zephyr/zephyr_project/modules/audio/sof/zephyr/../src/include/sof/drivers/timer.h:11,
                 from /home/osboxes/zephyr/zephyr_project/modules/audio/sof/src/lib/clk.c:10:
/home/osboxes/zephyr/zephyr_project/modules/audio/sof/zephyr/../src/include/sof/drivers/interrupt.h:11:10: fatal error: platform/drivers/interrupt.h: No such file or directory
   11 | #include <platform/drivers/interrupt.h>
```

**Environment (please complete the following information):**
 - OS:  Linux
 - Toolchain Zephyr SDK
 - Commit SHA: 536f0f5acbce4c974709d5c5ce224eaac6b08022


**Additional context**
In `sample.yaml` there is:
```
    integration_platforms:
      - intel_adsp_cavs15
```
So is it normal that twister try to build for other platforms ?
(I don't catch exactly the difference between `integration_platforms `and `platform_allow`)

However even when I do that, I get the following on intel_adsp_cavs15:

```
In file included from /home/galak/git/modules/audio/sof/src/platform/intel/cavs/platform.c:46:
/home/galak/git/modules/audio/sof/zephyr/../src/include/kernel/ext_manifest.h:32:10: fatal error: rimage/sof/kernel/ext_manifest.h: No such file or directory
   32 | #include <rimage/sof/kernel/ext_manifest.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/samples/audio/sof/sample.yaml
+++ b/samples/audio/sof/sample.yaml
@@ -7,5 +7,6 @@ tests:
   sample.audio.sof:
     tags: sof
     build_only: true
+    platform_allow: intel_adsp_cavs15 intel_adsp_cavs18 intel_adsp_cavs20
     integration_platforms:
       - intel_adsp_cavs15

--- a/scripts/ci/run_ci.sh
+++ b/scripts/ci/run_ci.sh
@@ -148,8 +148,6 @@ function west_setup() {
 		west init -l ${git_dir}
 		west update 1> west.update.log || west update 1> west.update-2.log
 		west forall -c 'git reset --hard HEAD'
-		# fetch sof submodules
-		west sof update
 	fi
 	popd
 }

--- a/soc/xtensa/intel_adsp/cavs_v15/Kconfig.defconfig.series
+++ b/soc/xtensa/intel_adsp/cavs_v15/Kconfig.defconfig.series
@@ -13,7 +13,7 @@ config SOC_TOOLCHAIN_NAME
 
 config SOC
 	string
-	default "intel_apl_adsp" if SOC_INTEL_CAVS_APL
+	default "intel_apl_adsp" if SOC_INTEL_CAVS_V15
 
 config SMP
 	default y

--- a/soc/xtensa/intel_adsp/cavs_v15/Kconfig.soc
+++ b/soc/xtensa/intel_adsp/cavs_v15/Kconfig.soc
@@ -4,7 +4,7 @@
 choice
 	prompt "Intel CAVS SoC Selection"
 
-	config SOC_INTEL_CAVS_APL
+	config SOC_INTEL_CAVS_V15
 		bool "Apollo Lake"
 		depends on SOC_SERIES_INTEL_CAVS_V15
 

--- a/west.yml
+++ b/west.yml
@@ -117,8 +117,7 @@ manifest:
       revision: 38c79a447e4a47d413b4e8d34448316a5cece77c
       path: modules/debug/segger
     - name: sof
-      west-commands: zephyr/west/west-commands.yml
-      revision: 1026dfdb13a3cf65d6a541c05c22a41df93c03d5
+      revision: ed0b4b930c40b1d0c044653658f6e9211848fdec
       path: modules/audio/sof
     - name: tinycbor
       path: modules/lib/tinycbor

